File: modern-java/UC-BerkeleyCS61B-Data-Structures/code/week1/worksheet/dog.java
///usr/bin/env jbang "$0" "$@" ; exit $?
//JAVA 25+

class Dog {
	String name;
	int size;

	Dog(String name, int size) {
		this.name = name;
		this.size = size;
	}

	void grow() {
		this.size += 1;
	}

	@Override
	public String toString() {
		return "%s the size %d dog".formatted(
				this.name,
				this.size);
	}
}

void main(String... args) {
	List<Dog> dogs = List.of(
			new Dog("maya", 1000),
			new Dog("yipster", 5),
			new Dog("scott", 25));

	System.out.println(dogs.get(0));
}

File: modern-java/UC-BerkeleyCS61B-Data-Structures/code/week1/worksheet/dog.py

class Dog:
    def __init__(self,name,size):
        self.name = name
        self.size = size
        
    def grow(self):
        self.size += 1
        

    def __str__(self):
        return f"{self.name} the size {self.size} dog"
    
dogs = [Dog("maya", 1000), Dog("yipster", 5), Dog("scott", 25)]
print(dogs[0])